### PR TITLE
fix: port provider bindgen fixes

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This commit fixes the provider bindgen issues for non http-server builds (ex. kv-redis)

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
LGTM, just needs a rebase it looks like 😄 Should just be the types import line in runtime

**Review by brooksmtownsend — APPROVED** [review]
Good to me as long as @rvolosatovs / @thomastaylor312 are cool with commenting out tests for now. I know that the tests are going to need a wider sweeping update to get them to properly test wRPC things, so in my opinion this is fine for iterating on a feature branch and we'll add tests back in before the PR to main.

File: crates/host/src/wasmbus/mod.rs [hunk 1]
```diff
@@ -177,6 +177,7 @@ impl Queue {
 #[derive(Debug)]
 struct ActorInstance {
     actor: wasmcloud_runtime::Component,
+    #[allow(unused)]
```
Comment (thomastaylor312) [inline crates/host/src/wasmbus/mod.rs:180]: Should all of these just be removed? Or are we just keeping them around for now? If we're going to remove in a follow up, let's leave a TODO comment so these are easier to find later

**Review by thomastaylor312 — APPROVED** [review]
I'm good with this. One minor comment. Also, I would think that adding the `#[ignore]` attribute to the tests is probably better than commenting the whole thing out and makes it easier to review when we reenable, but that is more of a nit

**Comment by vados-cosmonic** [comment]
@thomastaylor312 with regards to the ignoring the tests -- the tests don't compile at all because HTTP server is broken and no longer builable by the `build.rs` code -- that's why they're commented out.

File: crates/host/src/wasmbus/mod.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/host/src/wasmbus/mod.rs:180]: Will remove in a follow up!

**Pull request merged**